pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -212,3 +224,38 @@ func getKernelArgs(filepath string) (string, error) {
 	kernelArgs := liveISOArgMatch.ReplaceAllString(args.DefaultKernelArgs, "")
 	return kernelArgs, nil
 }
+
+func (a *AgentPXEFiles) handleAdditionals390xArtifacts(bootArtifactsFullPath, agentISOPath, arch string, initrdFile overlay.OverlayReader) error {
+	agentInitrdAddrFilePath := filepath.Join(a.tmpPath, "images", "initrd.addrsize")
+	addrsizeFile, err := isoeditor.NewInitrdAddrsizeReader(agentInitrdAddrFilePath, initrdFile)
+	if err != nil {
+		return err
+	}
+
+	agentInitrdAddrFile := filepath.Join(bootArtifactsFullPath, fmt.Sprintf("agent.%s-initrd.addrsize", a.cpuArch))
+	err = copyfile(agentInitrdAddrFile, addrsizeFile)
+	if err != nil {
+		return err
+	}
+
+	// with the size determination of initrd file the file pointer is at the end and need to be
+	// set to start again.
+	_, err = initrdFile.Seek(0, io.SeekStart)
```

Comment:
AFAICS, no else is consuming `a.ImageReader` after the [copyfile](https://github.com/openshift/installer/pull/8536/files#diff-38f93f2da042069046ec25c748458e4aeb5457951a609feec974083616298f42R93), so maybe this could be avoided now?

Reply:
Thank you, due to the copyfile the file pointer is at EOF and must be pushed to 0 again. Move code to the beginning.